src/models/event.js

```diff
@@ -1143,6 +1144,41 @@ utils.extend(MatrixEvent.prototype, {
     getTxnId() {
         return this._txnId;
     },
+
+    /**
+     * Get a copy/snapshot of this event. The returned copy will be loosely linked
+     * back to this instance, though will have "frozen" event information. Other
+     * properties may mutate depending on the state of this instance at the time
+     * of snapshotting.
+     *
+     * This is meant to be used to snapshot the event details themselves, not the
+     * features (such as sender) surrounding the event.
+     * @returns {MatrixEvent} A snapshot of this event.
+     */
+    getSnapshotCopy() {
```

Comment:
Maybe `toSnapshot`, similar to `toJSON`? `SnapshotCopy` seems redundant to me...

```suggestion
    toSnapshot() {
```